model/common/tests/common/states/unit_tests/test_factory.py

```diff
@@ -286,3 +289,29 @@ def test_composite_field_source_raises_upon_get_unknown_field(
     with pytest.raises(ValueError) as err:
         composite.get("alice")
         assert "not provided by source " in err.value  # type: ignore[operator]
+
+
+def reduce_scalar_min(ar: data_alloc.NDArray, xp: ModuleType) -> gtx.float:
+    while ar.ndim > 0:
+        ar = xp.min(ar)
```

Comment:
Also here, why is the while loop needed? If I understand correctly, if you leave out the `axis` parameter it should take the minimum over the whole (flattened) array, even if it has more than one dimension.